src/nunavut/lang/cpp/__init__.py

```diff
@@ -345,11 +521,17 @@ def filter_declaration(language: Language,
     if isinstance(instance, pydsdl.PrimitiveType) or isinstance(instance, pydsdl.VoidType):
         return filter_type_from_primitive(language, instance)
     elif isinstance(instance, pydsdl.VariableLengthArrayType):
-        return 'std::vector<{}>'.format(filter_declaration(language, instance.element_type))
+        variable_array_type = language.get_option('variable_array_type', DEFAULT_ARRAY_TYPE)
```

Comment:
The `DEFAULT_ARRAY_TYPE` does not seem to be an admissible default here because `std::array` cannot contain size information.

Reply:
https://en.cppreference.com/w/cpp/container/array/size ?

Reply:
What I mean is a runtime state: a variable-length DSDL array shall be resizable, but `std::array` is not.

Reply:
variable length arrays are vectors in this revision.

Reply:
That is true only as long as `variable_array_type` is explicitly defined, otherwise it's `std::array`. Maybe this line should rather raise an error if `variable_array_type` is not defined instead of defaulting to the fixed-size array?